src/linter.js

```diff
@@ -426,7 +426,11 @@ export default class Linter {
         // test runs against un-instrumented code.
         /* istanbul ignore if  */
         if (this.config.runAsBinary === true) {
-          process.exit(this.output.errors.length > 0 ? 1 : 0);
+          var exitCode = this.output.errors.length > 0 ? 1 : 0;
+          if (exitCode === 0 && this.config.warningsAsErrors === true) {
+            exitCode = this.output.warnings.length > 0 ? 1 : 0;
```

Comment:
I think this code might not be checked by our code coverage tool, but could you add a CLI test (I think there are existing ones for guidance) to make sure this works as it should? (Though the code looks fine.)